improved main

synarete opened the pull request:
Align `main` conventions with `samba-operator`. 
1) Log version and git-revision early on.
2) Helper function to create clientset.

synarete committed:
main: log process info upon boot

Log general info early on upon process boot; in particular, version and
git-revision info. Useful for debugging.

Fixed also minor revive warn: have comment to top-level main package.

Signed-off-by: Shachar Sharon <ssharon@redhat.com>

synarete committed:
main: create kubernetes.Clientset by helper function

Use helper function to create clientset for communication with k8s api
server. Try first to use in-cluster config, and only if fails use
out-of-cluster mode.

If user did not provide explicit KUBECONFIG, try to use the default
well-known location at "$HOME/.kube/config".

Signed-off-by: Shachar Sharon <ssharon@redhat.com>

synarete committed:
pkg: add packages comments

Make revive happy or else CI will fail.

Signed-off-by: Shachar Sharon <ssharon@redhat.com>

phlogistonjohn commented on main.go:
```diff
@@ -87,7 +89,7 @@ func main() {
 	if err != nil {
 		fmt.Printf("failed to set up logger\n")
-		os.Exit(2)
+		os.Exit(1)
```
This jumps out at me as not related to the rest of the patch. Is there a reason for this change?

synarete replied:
Try to have different exit code on different failures (1, 2 and 3) upon boot sequence. Maybe it deserves a dedicate PR for clarity.

phlogistonjohn replied:
IMO a separate patch/commit in this PR would be sufficient to make me happy. Along with that explanation in the commit msg. :-)

synarete replied:
Fair enough. Will fix that

synarete committed:
main: use different exit code upon failures

Use different exit-codes for failures upon boot sequence, so that
container failure cause may be detected quickly by its exit code.

Signed-off-by: Shachar Sharon <ssharon@redhat.com>

phlogistonjohn reviewed (approved):
LGTM
